libpod/image.go

```diff
@@ -216,3 +217,45 @@ func (r *Runtime) GetImages(filter ...ImageFilter) ([]*storage.Image, error) {
 func (r *Runtime) ImportImage(path string) (*storage.Image, error) {
 	return nil, ctr.ErrNotImplemented
 }
+
+// FindImage searches for a *storage.Image with a matching the given name or ID in the given store.
+func (r *Runtime) FindImage(image string) (*storage.Image, error) {
+	var img *storage.Image
+	ref, err := is.Transport.ParseStoreReference(r.store, image)
+	if err == nil {
+		img, err = is.Transport.GetStoreImage(r.store, ref)
+	}
+	if err != nil {
+		img2, err2 := r.store.Image(image)
+		if err2 != nil {
+			if ref == nil {
+				return nil, errors.Wrapf(err, "error parsing reference to image %q", image)
+			}
+			return nil, errors.Wrapf(err, "unable to locate image %q", image)
+		}
+		img = img2
+	}
+	return img, nil
+}
+
+// FindImageRef searches for and returns a new types.Image matching the given name or ID in the given store.
+func (r *Runtime) FindImageRef(image string) (types.Image, error) {
+	img, err := r.FindImage(image)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to locate image %q", image)
+	}
+	ref, err := is.Transport.ParseStoreReference(r.store, "@"+img.ID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing reference to image %q", img.ID)
+	}
+	imgRef, err := ref.NewImage(nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading image %q", img.ID)
+	}
+	return imgRef, nil
+}
+
+// SetNames Calls the store SetNames function to set the names of the image
+func (r *Runtime) SetNames(id string, names []string) error {
```

Comment:
This should be changed to TagImage, to better describe what it does. I think we want to eventually make it handle the entirety of the tagging process (move the stuff that's in cmd/kpod in here), but that can come later.